### PR TITLE
Add support for protocols and file://

### PR DESCRIPTION
**Pull request by MotoLegacy**
Allows for specifying a file to open via command line arguments and using the new protocol backend for supporting the `file://` protocol. Sets up some ground work to be able to easily implement HTTP(S) retrieval of documents without much trouble.

`./pantomime file://tests/simple_document.html` as an example of usage.

### CODE REVIEW

File: source/prtcl/prtcl_file.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+    // Seek to the end of the file to get its size.
+    fseek(document, 0, SEEK_END);
+    int document_size = ftell(document);
+    fseek(document, 0, SEEK_SET);
+
+    // Toss the HTML data into a char[].
+    char html_doc[document_size];
+    int i = -1;
+    int c;
+    while(EOF != (c = fgetc(document)) && ++i < document_size)
```
Comment (krazynez) [inline source/prtcl/prtcl_file.c]: why not use fread here instead?

File: source/prtcl/prtcl_main.c [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+
+#include <util.h>
+#include <prtcl/protocol_file.h>
+
+char* PRTCL_RetrieveDocument(char* req_uri)
+{
+    // Check the protocol identifier so we know
+    // how to handle retrieving the document.
+    // file://
+    if (req_uri[0] == 'f' && req_uri[1] == 'i' &&
```
Comment (krazynez) [inline source/prtcl/prtcl_main.c:28]: This works yes, but will get messy the more protocols we support I feel.

**Review by krazynez — APPROVED** [review]
Looks good, just a few questions really.

File: source/prtcl/prtcl_main.c (hunk 2 above)
Comment (MotoLegacy) [reply source/prtcl/prtcl_main.c:28]: Can you suggest an alternative? Protocols have inconsistent lengths so we cannot just build a chunk at whatever coordinate. So the only other solution I can fathom would be to walk through the string first to find "://" to know the index at which to end the split, which sounds a lot less elegant than what is being done here.

**Pull request merged**